HTTP API test improvements (#2315)

wcs1only opened the pull request:
# Description

We now have tests exercising every HTTP error status code. 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2315

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

wcs1only commented:
/ok-to-test

mukundansundar reviewed (commented):
https://github.com/dapr/dapr/pull/2344/files#diff-237f098772824bb7f3c25e36c2565bf147e4dfedb1077bb351595b0ba4b881bdL490 should not be unauthorized?

ERR_ACTOR_RUNTIME_NOT_FOUND should be ERR_ACTOR_RUNTIME_NOT_CONFIGURED?
consequently should all the NOT_FOUND when a.publishFn == nil or a.actor == nil etc be changed to NOT_CONFIGURED similar to when a.stateStores == nil?

wcs1only commented:
> https://github.com/dapr/dapr/pull/2344/files#diff-237f098772824bb7f3c25e36c2565bf147e4dfedb1077bb351595b0ba4b881bdL490 should not be unauthorized?
> 
> ERR_ACTOR_RUNTIME_NOT_FOUND should be ERR_ACTOR_RUNTIME_NOT_CONFIGURED?
> consequently should all the NOT_FOUND when a.publishFn == nil or a.actor == nil etc be changed to NOT_CONFIGURED similar to when a.stateStores == nil?

I'm going to pick up a full audit of the error messages as part of #2338. For now, just trying to get the coverage up. My plan is to increase coverage -> audit error codes/messages -> update the tests to reflect the behavior we want -> let the tests tell us where the code needs to change -> fix the code.

mukundansundar reviewed (commented):
https://github.com/dapr/dapr/pull/2344/files#diff-237f098772824bb7f3c25e36c2565bf147e4dfedb1077bb351595b0ba4b881bdL337

https://github.com/dapr/dapr/pull/2344/files#diff-237f098772824bb7f3c25e36c2565bf147e4dfedb1077bb351595b0ba4b881bdL475

Should these be 204? 
L337 L475 L699 L809 L982


Should  L636 be respondEmpty(reqCtx, fasthttp.StatusCreated) since it is `onCreateActorReminder` and L671 also in `onCreateActorTimer` similar to `onPostState` method?

Should L932 be StatusCreated for `onPutMetadata`?

wcs1only commented:
> https://github.com/dapr/dapr/pull/2344/files#diff-237f098772824bb7f3c25e36c2565bf147e4dfedb1077bb351595b0ba4b881bdL337
> 
> https://github.com/dapr/dapr/pull/2344/files#diff-237f098772824bb7f3c25e36c2565bf147e4dfedb1077bb351595b0ba4b881bdL475
> 
> Should these be 204?
> L337 L475 L699 L809 L982

Added to the table in #2338

> Should L636 be respondEmpty(reqCtx, fasthttp.StatusCreated) since it is `onCreateActorReminder` and L671 also in `onCreateActorTimer` similar to `onPostState` method?

Already there in #2338  👍 

> Should L932 be StatusCreated for `onPutMetadata`?

Added to #2338

wcs1only commented:
/ok-to-test

mukundansundar commented on pkg/http/api_test.go:
```diff
@@ -583,6 +647,83 @@ func TestV1ActorEndpoints(t *testing.T) {
+			ActorID:   "fakeActorID",
+			ActorType: "fakeActorType",
+			Key:       "key1",
+		}).Return(nil, errors.New("UPSTREAM_ERROR"))
```
FYI - can use `assert.AnError` in place of creating a new error value for tests.

wcs1only replied:
TIL - Thanks!
